spec/requests/search_spec.rb

```diff
@@ -166,61 +166,108 @@
+          mock_documents,
+        )
 
-      get "/search/feed_content"
-      expect(Search::User).to have_received(:search_documents)
-      expect(Search::FeedContent).to have_received(:search_documents)
-    end
+        get search_feed_content_path
+        expect(Search::User).to have_received(:search_documents)
+        expect(Search::FeedContent).to have_received(:search_documents)
+      end
 
-    it "queries for only Articles and Podcast Episodes if class_name!=User" do
-      allow(Search::FeedContent).to receive(:search_documents).and_return(
-        mock_documents,
-      )
-      allow(Search::User).to receive(:search_documents)
+      it "queries for only Articles and Podcast Episodes if class_name!=User" do
+        allow(Search::FeedContent).to receive(:search_documents).and_return(
+          mock_documents,
+        )
+        allow(Search::User).to receive(:search_documents)
 
-      get "/search/feed_content?class_name=Article"
-      expect(Search::User).not_to have_received(:search_documents)
-      expect(Search::FeedContent).to have_received(:search_documents)
+        get search_feed_content_path(class_name: "Article")
+        expect(Search::User).not_to have_received(:search_documents)
+        expect(Search::FeedContent).to have_received(:search_documents)
+      end
+
+      it "queries for approved" do
+        allow(Search::FeedContent).to receive(:search_documents).and_return(
+          mock_documents,
+        )
+
+        get search_feed_content_path(class_name: "Article", approved: "true")
+        expect(Search::FeedContent).to have_received(:search_documents).with(
+          params: { "approved" => "true", "class_name" => "Article" },
+        )
+      end
     end
 
-    it "queries for approved" do
-      allow(Search::FeedContent).to receive(:search_documents).and_return(
-        mock_documents,
-      )
+    context "when using PostgreSQL" do
+      before do
+        allow(FeatureFlag).to receive(:enabled?).with(:search_2_homepage).and_return(true)
+      end
 
-      get "/search/feed_content?class_name=Article&approved=true"
-      expect(Search::FeedContent).to have_received(:search_documents).with(
-        params: { "approved" => "true", "class_name" => "Article" },
-      )
+      it "does not call Homepage::FetchArticles with the incorrect combination of params", :aggregate_failures do
+        allow(Homepage::FetchArticles).to receive(:call)
+
+        get search_feed_content_path
+        expect(Homepage::FetchArticles).not_to have_received(:call)
+
+        get search_feed_content_path(class_name: "Article", search_fields: "keyword")
+        expect(Homepage::FetchArticles).not_to have_received(:call)
+      end
+
+      it "calls Homepage::FetchArticles with the correct combination of params" do
+        allow(Homepage::FetchArticles).to receive(:call)
+
+        get search_feed_content_path(class_name: "Article")
+
+        expect(Homepage::FetchArticles).to have_received(:call)
```

Comment:
This expectation doesn't assert what the spec title says it does

Reply:
I updated the description